core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -65,77 +65,98 @@ public PointList simplify() {
+        for (int i = 0; i < pathSegmentations; i++) {
+            currIntervalEnd[i] = getLength(this.pathSegmentations.get(i), currIntervalIndex[i]);
+        }
+
+        // to correctly update the interval indices in path details and instructions we need to keep track of how
+        // many points were removed by Douglas-Peucker in the current and previous intervals
+        final int[] removedPointsInCurrInterval = new int[pathSegmentations];
+        final int[] removedPointsInPrevIntervals = new int[pathSegmentations];
+
+        for (int p = 0; p < pointList.size(); p++) {
+            int removed = 0;
+            // first we check if we hit an interval end for one of the segmenations and run douglas peucker if so
+            for (int s = 0; s < pathSegmentations; s++) {
+                if (segmentsFinished[s]) {
+                    continue;
                 }
-                if (startIntervals[i] >= nonConflictingStart && endIntervals[i] <= nonConflictingEnd) {
-                    listIndexToSimplify = i;
+                if (p == currIntervalEnd[s]) {
+                    // This is important for performance: we must not compress the point list after each call to
+                    // simplify, otherwise a lot of data is copied, especially for long routes (e.g. many via nodes),
+                    // see #1764. Note that since the point list does not get compressed we have to keep track of the
+                    // total number of removed points to shift the indices into pointList correctly.
+                    final boolean compress = false;
+                    removed = douglasPeucker.simplify(pointList, intervalStart, currIntervalEnd[s], compress);
+                    intervalStart = p;
+                    break;
                 }
             }
-
-            if (listIndexToSimplify >= 0 && simplificationPossible) {
-                // Only simplify if there is more than one point
-                if (nonConflictingEnd - nonConflictingStart > 1) {
-                    int removed = douglasPeucker.simplify(pointList, nonConflictingStart, nonConflictingEnd);
-                    if (removed > 0) {
-                        for (int i = 0; i < listsToSimplify.size(); i++) {
-                            List pathDetails = listsToSimplify.get(i);
-                            reduceLength(pathDetails, offsets[i], startIntervals[i], endIntervals[i] - removed);
-                            // This is not needed for Instructions, as they don't contain references, but PointLists
-                            if (pathDetails.get(0) instanceof PathDetail) {
-                                for (int j = offsets[i] + 1; j < pathDetails.size(); j++) {
-                                    PathDetail pd = (PathDetail) pathDetails.get(j);
-                                    reduceLength(pathDetails, j, pd.getFirst() - removed, pd.getLast() - removed);
-                                }
-                            }
```

Comment:
This code took most of the time after I fixed the `compressNew` bug in the first commit